src/Symfony/Component/DependencyInjection/ServiceLocator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection;
+
+use Psr\Container\ContainerInterface as PsrContainerInterface;
+use Symfony\Component\DependencyInjection\Exception\ServiceCircularReferenceException;
+use Symfony\Component\DependencyInjection\Exception\ServiceNotFoundException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in version 3.3
+ */
+class ServiceLocator implements PsrContainerInterface, \IteratorAggregate
```

Comment:
This implementation of PSR-11 looks in fact quite nice standalone to me, no need to have it tightly coupled with the way one can use config to create one. It's already useful on it own.